src/Bicep.Core/Configuration/ModuleAliasesConfiguration.cs

```diff
@@ -1,50 +1,142 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Bicep.Core.Modules;
 using Microsoft.Extensions.Configuration;
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using static Bicep.Core.Diagnostics.DiagnosticBuilder;
 
 namespace Bicep.Core.Configuration
 {
-    public record TemplateSpecModuleAlias
+    public record ModuleAlias { }
+
+    public record TemplateSpecModuleAlias : ModuleAlias
     {
         public string? Subscription { get; init; }
 
         public string? ResourceGroup { get; init; }
+
+        public override string ToString()
+        {
+            Debug.Assert(this.Subscription is not null, "TemplateSpecModuleAlias.Subscription should already be validated and cannot be null.");
```

Comment:
>Assert [](http://example.com/codeflow?start=18&length=6)

`ToString()` shouldn't expect any specific calling pattern that prevents the assertion from fail. (The record/class should be independent.)

Reply:
Same below.

Reply:
Good pt. Removed the assertions.